agent/internal/actor/actor.go

```diff
@@ -76,19 +76,30 @@ func (s *Store) FindIP(ip net.IP) (action Action, exists bool, err error) {
 	}
 
 	if stdIPv4 := ip.To4(); stdIPv4 != nil {
-		if store.treeV4 == nil {
+		tree := store.treeV4
+		if tree == nil {
 			return nil, false, nil
 		}
-		IPv4 := patricia.NewIPv4AddressFromBytes(stdIPv4, 32)
-		action, err := store.treeV4.findAction(&IPv4)
-		if err != nil {
-			return nil, false, err
+		IPv4 := patricia.NewIPv4AddressFromBytes(stdIPv4, ipv4Bits)
+		action, err = tree.findAction(&IPv4)
+	} else if stdIPv6 := ip.To16(); stdIPv6 != nil {
+		// warning: the previous condition is also true with ipv4 address (as they
+		// can be represented using ipv6 ::ffff:ipv4), so testing the ipv4 first is
+		// important to avoid entering this case even with ipv6 addresses.
+		tree := store.treeV6
+		if tree == nil {
+			return nil, false, nil
 		}
-		// action may be nil if ip does not exist in the tree.
-		return action, action != nil, nil
+		IPv6 := patricia.NewIPv6Address(stdIPv6, ipv6Bits)
+		action, err = tree.findAction(&IPv6)
```

Comment:
What if it's neither ipv4 or ipv6?

Reply:
The `To16()` and `To4()` methods return `nil`. So we fall into the `return action, action != nil, nil` which gives `return nil, false, nil`.

Reply:
lgtm